pkg/scheduler/objects/application.go

```diff
@@ -2010,6 +2033,15 @@ func (sa *Application) CleanupUsedResource() {
 	sa.usedResource = nil
 }
 
+func (sa *Application) CleanupPreemptedResource() {
+	sa.preemptedResource = nil
+}
+
+func (sa *Application) CleanupTrackedResource() {
+	sa.CleanupUsedResource()
+	sa.CleanupPreemptedResource()
+}
```

Comment:
We need to be careful about these methods.

For example, right now `CleanupUsedResource()` is called from two places: `moveTerminatedApp()` and the state machine. One path has the lock of the `Application` but the other one doesn't.

We need to be consistent with this, modification must occur under a write lock. We generally have two methods in these cases: an exported and an unexported. The exported calls lock/defer unlock and invokes the unexported which doesn't deal with locks. 

Please make sure that we follow this pattern and modification is safe.

Reply:
Thanks, good point! wonder if you prefer which option below:

Option 1: make the methods private: 
func (sa *Application) cleanupUsedResource() {
	sa.usedResource = nil
}

func (sa *Application) cleanupPreemptedResource() {
	sa.preemptedResource = nil
}

func (sa *Application) CleanupTrackedResource() {
	sa.cleanupUsedResource()
	sa.cleanupPreemptedResource()
}

Option 2: Only use one method:
func (sa *Application) CleanupTrackedResource() {
  sa.usedResource = nil
  sa.preemptedResource = nil
}